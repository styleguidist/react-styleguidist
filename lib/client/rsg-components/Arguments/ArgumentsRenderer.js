import "core-js/modules/es.array.map";
import "core-js/modules/es.function.name";
import "core-js/modules/es.object.assign";

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Argument from 'rsg-components/Argument';
import Heading from 'rsg-components/Heading';
export var styles = function styles(_ref) {
  var space = _ref.space;
  return {
    root: {
      marginBottom: space[2],
      fontSize: 'inherit'
    },
    headingWrapper: {
      marginBottom: space[0]
    }
  };
};
export var ArgumentsRenderer = function ArgumentsRenderer(_ref2) {
  var classes = _ref2.classes,
      args = _ref2.args,
      heading = _ref2.heading;

  if (args.length === 0) {
    return null;
  }

  return /*#__PURE__*/React.createElement("div", {
    className: classes.root
  }, heading && /*#__PURE__*/React.createElement("div", {
    className: classes.headingWrapper
  }, /*#__PURE__*/React.createElement(Heading, {
    level: 5
  }, "Arguments")), args.map(function (arg) {
    return /*#__PURE__*/React.createElement(Argument, _extends({
      key: arg.name
    }, arg));
  }));
};
ArgumentsRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  args: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    type: PropTypes.object,
    description: PropTypes.string
  }).isRequired).isRequired,
  heading: PropTypes.bool
};
export default Styled(styles)(ArgumentsRenderer);