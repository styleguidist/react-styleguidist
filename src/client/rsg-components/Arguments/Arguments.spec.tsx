import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import { ArgumentsRenderer, styles } from './ArgumentsRenderer';

const props = {
	classes: classes(styles),
};

const args = [
	{
		name: 'Foo',
		description: 'Converts foo to bar',
		type: { name: 'Array' },
	},
	{
		name: 'Foo',
	},
];

it('renderer should render arguments', () => {
	const renderer = createRenderer();
	renderer.render(<ArgumentsRenderer {...props} args={args} />);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('renderer should render heading', () => {
	const renderer = createRenderer();
	renderer.render(<ArgumentsRenderer {...props} args={[args[1]]} heading />);

	expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('renderer should render nothing for empty array', () => {
	const renderer = createRenderer();
	renderer.render(<ArgumentsRenderer {...props} args={[]} />);

	expect(renderer.getRenderOutput()).toBe(null);
});
