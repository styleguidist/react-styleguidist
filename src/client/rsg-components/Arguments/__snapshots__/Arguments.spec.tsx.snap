// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render arguments 1`] = `
<div
  className="root"
>
  <Styled(Argument)
    description="Converts foo to bar"
    name="Foo"
    type={
      Object {
        "name": "Array",
      }
    }
  />
  <Styled(Argument)
    name="Foo"
  />
</div>
`;

exports[`renderer should render heading 1`] = `
<div
  className="root"
>
  <div
    className="headingWrapper"
  >
    <Styled(Heading)
      level={5}
    >
      Arguments
    </Styled(Heading)>
  </div>
  <Styled(Argument)
    name="Foo"
  />
</div>
`;
